### PR TITLE
Set statement_timeout to 0 when running migrations

### PR DESCRIPTION
**Pull request by oliverguenther**
Add a db:migrate rake task hook to run migrations with a statement_timeout of zero.

### CODE REVIEW

**Comment by machisuji** [comment]
@oliverguenther I guess [this](https://github.com/opf/openproject/blob/release/13.0/db/migrate/20230608151123_add_validity_period_to_journals.rb#L2C3-L2C32) can be removed then.

File: lib/tasks/database.rake [hunk 1]
```diff
@@ -97,7 +97,12 @@ namespace 'openproject' do
...
     end
+
+    task remove_statement_timeout: %w[openproject:db:check_connection] do
+      ActiveRecord::Base.connection.execute("SET statement_timeout = 0;")
+    end
   end
 end
 
 Rake::Task["db:migrate"].enhance ["openproject:db:ensure_database_compatibility"]
+Rake::Task["db:migrate"].enhance ["openproject:db:remove_statement_timeout"]
```
Comment (ulferts) [inline lib/tasks/database.rake:108]: Seems to be a good solution for the problem 👍

File: docs/installation-and-operations/operation/upgrading/README.md [hunk 2]
```diff
@@ -8,14 +8,13 @@ sidebar_navigation:
...
-| [Troubleshooting](#troubleshooting)                          | Solutions for common problems                                |
-| [Upgrade notes to 13.x](#upgrade-notes-to-13x)               | How to upgrade from OpenProject 9.x or greater to OpenProject 12.x. |
-| [Upgrade notes for 8.x to 9.x](#upgrade-notes-for-8x-to-9x)  | How to upgrade from OpenProject 8.x to OpenProject 9.x.      |
-| [Upgrade notes for 7.x to 8.x](#upgrade-notes-for-openproject-7x-to-8x) | How to upgrade from OpenProject 7.x to OpenProject 8.x.      |
+| Topic                                                        | Content                                                     |
+| ------------------------------------------------------------ | ----------------------------------------------------------- |
+| [Package-based installation](#package-based-installation-debrpm) | How to upgrade a package-based installation of OpenProject. |
+| [Docker-based installation](#compose-based-installation)      | How to upgrade a Docker-based installation of OpenProject.  |
+| [Upgrade notes to 12.x](#upgrade-notes-to-12x)  | How to upgrade from OpenProject 9.x or greater to OpenProject 12.x.     |
```
Comment (cbliard) [inline docs/installation-and-operations/operation/upgrading/README.md]: There is no "Upgrade notes to 13.x" anymore. Is it intended?

Comment (ulferts) [reply docs/installation-and-operations/operation/upgrading/README.md]: We could alter the `Upgrade notes to 12.x` to more clearly include 13.0. Maybe something like `Upgrade notes from 9.x` which we can keep until something special comes along.

**Pull request merged**